### PR TITLE
Automatic SMS Verification with the SMS Retriever API in Android

### LINKED ISSUE #3520: Automatic SMS Verification with the SMS Retriever API in Android
### Introduction paragraph (2-3 paragraphs):
With the SMS Retriever API, you can perform SMS-based user verification in your Android app automatically, without requiring the user to manually type OTP codes, and without requiring any extra app permissions.
This makes their on-boarding feel seamless and kind of effortless on the part of the user. When the user's device receives the SMS message, Google Play services uses the app hash to determine that the message is intended for your app, and makes the message text available to your app through the SMS Retriever API. This would be a good feature that would differentiate your app from other apps that force users to enter it manually. 

### Key takeaways: 
-  What is SMS Verification process
-  How to implement automatic SMS verification feature on your app.

### References: 
https://developers.google.com/

### PR DESCRIPTION
**Pull request by robert-muriithi**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes  #3520

### CODE REVIEW

**Comment by robert-muriithi** [comment]
Hello @ahmadmardeni1 and @LinusMuema . The plagiarism in my article is brought about by the code I wrote. Firstly, some codes are common in all applications written in kotlin. In my case, code such as "class MainActivity : AppCompatActivty(){}" and its body are commonly used in android applications are flagged as plagiarised code. The second thing that is raising the plagiarism level is the `AppSignatureHelper class`. This class is provided by Google and it is the one used to generate the app's hash string. It is documented on google documentation and is used by many. The problem is that the whole code is detected as plagiarized code since it has been used in different blogs and sites that try to explain this concept. The code is the same all over. What should I  do about it now that Copyleaks is flagging my code as plagiarized code?

File: content/articles/automatic-sms-verification-with-the-sms-retriever-api-in-android/index.md [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+
+Once you obtain the user's mobile number from the Edit text, send it to the server perfoming the verification. The server will send back a code that to the number that was sent to it. This now will have verified the phone number to your application. To perform SMS verification on a server, check out [SMS Verification on Server]. (https://developers.google.com/identity/sms-retriever/verify)
+
+### Step 4: Getting an instance of the SmsRetriverClient object
+
+We'll first create an instance of the SmsRetrieverClient object. This is followed by invoking its initSmsRetriever instance function and adding `onSuccessListener` and `onFailureListener` to the task. We wrap all these code in a function for later use.
+
+```
+private fun initSmsListener() {
```
Comment (Ericgacoki) [inline content/articles/automatic-sms-verification-with-the-sms-retriever-api-in-android/index.md:61]: Did you forget something?
Oh, yes! You forgot to add the `language identifier` in your code snippets.

**Review by Ericgacoki — CHANGES_REQUESTED** [review]
Hey @robert-muriithi, you've covered a great concept in the Android context. However, fix the following minor issue before I proceed with the review!

**Comment by Ericgacoki** [comment]
A few more snippets to go!

File: content/articles/automatic-sms-verification-with-the-sms-retriever-api-in-android/index.md [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+        }
+}
+
+```
+
+The above function is invoked on the `onCreate()` method.
+
+```
+class MainActivity : AppCompatActivity() {
+    private lateinit var  binding: ActivityMainBinding
```
Comment (Ericgacoki) [inline content/articles/automatic-sms-verification-with-the-sms-retriever-api-in-android/index.md:104]: You seem to have omitted this and several others. I'll fix them for you but next time be sure to make the requested changes where applicable.

Comment (robert-muriithi) [reply content/articles/automatic-sms-verification-with-the-sms-retriever-api-in-android/index.md:104]: Okay thanks @Ericgacoki . Incase of any other changes i will rectify

**Comment by hectorkambow** [comment]
> Thanks for explaining @robert-muriithi Don't worry about the code, just change the other highlighted sentences.

Only if its boilerplate type code - any other code snippets should be completely ORIGINAL 👍 

cc @robert-muriithi

**Pull request closed without merging**